src/history/historyTracker.ts

```diff
@@ -93,6 +93,50 @@ class HistoryStep {
     this.cursorStart = init.cursorStart || undefined;
     this.marks     = init.marks || [];
   }
+
+  /**
+   * merge collapses individual character changes into larger blocks of changes
+   */
+  public merge(): void {
+    if (this.changes.length < 2) {
+      return;
+    }
+    // create new changes list
+    var merged: DocumentChange[] = [];
+    // iterate through the changes, saving the first outside the loop
+    var iter = (this.changes[Symbol.iterator]());
```

Comment:
Yes I considered that at first, but then on the first iteration `prev` and `change` would be the same object. `change` needs to skip the first element, which is why I make an iterator, get the first element, then iterate through the rest.

Other options include using `var prev = this.changes.splice(0, 1)[0];` or maybe `var prev = this.changes[0]; for (const next of this.changes.slice(1))`. I'd prefer the slice method just because it doesn't alter the original array.

I went with the manual iterator because it doesn't change the original array, it only loops through the array once, and it doesn't create another object. I fully admit these are quite arbitrary reasons, two of which pretty clearly premature optimization. :smile: 

What would you prefer?